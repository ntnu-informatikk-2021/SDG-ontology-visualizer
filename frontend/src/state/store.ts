import { createStore } from 'redux';
import apiErrorReducer from './reducers/apiErrorReducer';

const store = createStore(apiErrorReducer);

export default store;
