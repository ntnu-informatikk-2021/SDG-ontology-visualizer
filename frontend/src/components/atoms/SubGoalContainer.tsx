import {
  Accordion,
  AccordionButton,
  AccordionIcon,
  AccordionItem,
  AccordionPanel,
  Heading,
  Text,
} from '@chakra-ui/react';
import React from 'react';
import { useSelector } from 'react-redux';
import colorSwitcher from '../../common/colorSwitcher';
import { RootState } from '../../state/store';
import { SubGoal } from '../../types/ontologyTypes';

type SubGoalContainerProps = {
  subGoalNode: SubGoal;
};

const SubGoalContainer: React.FC<SubGoalContainerProps> = ({
  subGoalNode,
}: SubGoalContainerProps) => {
  const selectedNode = useSelector((state: RootState) => state.ontology.selectedNode);

  return (
    <Accordion allowToggle>
      <AccordionItem boxShadow="lg" borderRadius="md">
        <AccordionButton
          _expanded={{ borderBottomRadius: '0' }}
          borderRadius="md"
          bg={colorSwitcher(selectedNode! && selectedNode.id)}
          color="white"
          _hover={{ opacity: '75%' }}
        >
          <Heading as="h3" size="sm">
            {subGoalNode.SubjectLabel}
          </Heading>
          <AccordionIcon />
        </AccordionButton>
        <AccordionPanel>
          <Text fontSize="sm">{subGoalNode.description}</Text>
        </AccordionPanel>
      </AccordionItem>
    </Accordion>
  );
};

export default SubGoalContainer;
