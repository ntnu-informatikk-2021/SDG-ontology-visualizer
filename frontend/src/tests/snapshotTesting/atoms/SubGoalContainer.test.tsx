import React from 'react';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import SubGoalContainer from '../../../components/atoms/SubGoalContainer';
import store from '../../../state/store';

const subGoalNode1 = {
  id: 't1',
  name: 'hddei',
  type: 'okd1',
  prefix: {
    prefix: 'jdajaa',
    iri: '123d4',
  },
  correlation: 5,
  SubjectLabel: 'eeat',
  description: 'meee',
};

it('renders when there is one item', () => {
  const tree = renderer
    .create(
      <Provider store={store}>
        <SubGoalContainer subGoalNode={subGoalNode1} />
      </Provider>,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
