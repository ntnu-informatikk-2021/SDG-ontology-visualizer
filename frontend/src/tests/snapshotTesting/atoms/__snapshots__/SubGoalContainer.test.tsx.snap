// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders when there is one item 1`] = `
<div
  className="chakra-accordion css-0"
>
  <div
    className="chakra-accordion__item css-rydg3k"
  >
    <button
      aria-controls="accordion-panel-1"
      aria-expanded={false}
      className="chakra-accordion__button css-9rvr9d"
      disabled={false}
      id="accordion-button-1"
      onClick={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      type="button"
    >
      <h3
        className="chakra-heading css-0"
      >
        eeat
      </h3>
      <svg
        aria-hidden={true}
        className="chakra-icon chakra-accordion__icon css-8o5ky2"
        focusable={false}
        viewBox="0 0 24 24"
      >
        <path
          d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z"
          fill="currentColor"
        />
      </svg>
    </button>
    <div
      className="chakra-collapse"
      style={
        Object {
          "display": "none",
          "height": "0px",
          "opacity": 0,
          "overflow": "hidden",
        }
      }
    >
      <div
        aria-labelledby="accordion-button-1"
        className="chakra-accordion__panel css-0"
        id="accordion-panel-1"
        role="region"
      >
        <p
          className="chakra-text css-g5hezf"
        >
          meee
        </p>
      </div>
    </div>
  </div>
</div>
`;
